cli/src/main.rs

```diff
@@ -0,0 +1,221 @@
+use clap::Parser;
+use object_store::{memory::InMemory, path::Path, ObjectStore};
+use rustyline::error::ReadlineError;
+use rustyline::DefaultEditor;
+use shell_words::split;
+use slatedb::config::{CompactorOptions, DbOptions, ObjectStoreCacheOptions};
+use slatedb::db::Db;
+use slatedb::inmemory_cache::InMemoryCacheOptions;
+use std::{sync::Arc, time::Duration};
+use tokio;
+
+#[derive(Parser, Debug)]
+#[command(name = "SlateDB REPL CLI")]
+#[command(version = "1.0")]
+#[command(about = "A command-line interface for SlateDB", long_about = None)]
+struct Args {
+    /// Path to the database storage directory
+    #[arg(short, long, value_name = "PATH", default_value = "/tmp/test_kv_store")]
+    storage_path: String,
+}
+
+// this is the REPL
+struct SlateDbREPL {
+    db: Db,
+}
+
+impl SlateDbREPL {
+    /// Initialize the DatabaseCLI with a connected database
+    async fn new(storage_path: &str) -> Result<Self, Box<dyn std::error::Error>> {
+        let object_store: Arc<dyn ObjectStore> = Arc::new(InMemory::new());
+        let options = DbOptions {
+            flush_interval: Duration::from_millis(100),
+            manifest_poll_interval: Duration::from_millis(100),
+            min_filter_keys: 10,
+            l0_sst_size_bytes: 128,
+            l0_max_ssts: 8,
+            max_unflushed_memtable: 2,
+            compactor_options: Some(CompactorOptions::default()),
+            compression_codec: None,
+            block_cache_options: Some(InMemoryCacheOptions::default()),
+            object_store_cache_options: ObjectStoreCacheOptions::default(),
+        };
+        let db = Db::open_with_opts(Path::from(storage_path), options, object_store).await?;
+        Ok(SlateDbREPL { db })
+    }
+
+    async fn put(&self, key: &str, value: &str) -> Result<(), Box<dyn std::error::Error>> {
+        self.db.put(key.as_bytes(), value.as_bytes()).await;
+        Ok(())
+    }
```

Comment:
`self.db.put` doesn't return any error so this can be simplified to just one line

```suggestion
    async fn put(&self, key: &str, value: &str) {
        self.db.put(key.as_bytes(), value.as_bytes()).await
    }
```

in case this was done just to have a consistent return type in all the API methods then i guess its ok, I am very new to Rust so I don't have strong opinions about it just pointing it out